### DISCUSSION TITLE
[#1065647] API Suggestion: Close/Open/Edit Panes and their content

### DISCUSSION THREAD

**Discussion by Hatchibombotar** [Mojang/minecraft-editor#123] [Feedback]
The api should have the ability to modify panes and their elements after they are created. For example, you may want create an editor extension that has multiple modes, and options that only work in some of the modes, or a tool where the elements of a dropdown are dynamic. I don't believe this is currently possible for the undocumented server-editor 0.3.0 module.

**Comment by dacowan** [comment]
It's possible, but tricky.  I'm working on a number of samples that demonstrate this functionality.
However, I'm well aware that it's a pain point (it gets on my nerves here too) so I'm looking to extend the dynamic nature of the UI elements so that binding is much easier and transparent.
I can't tell you when those samples will be ready, but it won't be too much longer - just hang in there, I know it's frustrating.

**Comment by conmaster2112** [comment]
> The api should have the ability to modify panes and their elements after they are created. For example, you may want create an editor extension that has multiple modes, and options that only work in some of the modes, or a tool where the elements of a dropdown are dynamic. I don't believe this is currently possible for the undocumented server-editor 0.3.0 module.

I think, if this every come to reality then client APIs is they way, managing so many client things via server is not good for performance
